sklearn/metrics/_classification.py

```diff
@@ -274,6 +274,9 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None,
 
     if labels is None:
         labels = unique_labels(y_true, y_pred)
```

Comment:
So we should have something like this. We will need a test to ensure that `labels=[]` raise the proper error.
```suggestion
        labels = unique_labels(y_true, y_pred)
    else:
        labels = np.asarray(labels)
        n_labels = label.size
        if n_labels == 0:
            raise ValueError("Forbidden")
        if y_true.size == 0:
            return np.zeros(...)
        if np.all([l not in y_true for l in labels]):
            raise ValueError("At least one label specified must be in y_true")
```